aquadoggo/src/schema/static_schema_provider.rs

```diff
@@ -0,0 +1,194 @@
+/// let ptr_leaked = leaked as *const i32;
+/// assert_eq!(ptr_owned, ptr_leaked);
+///
+/// // Drop the reference to that value. Since we consumed the box already Rust doesn't manage the
+/// // memory anymore for us, it is still there! We have a memory leak.
+/// drop(leaked);
+///
+/// // Let's look into whats at that memory location, and indeed, we can still read the value even
+/// // though all references are gone .. :-(
+/// let points_at = unsafe { *ptr_leaked };
+/// assert_eq!(points_at, 100);
+/// ```
+///
+/// See `StaticLeak` tests for more examples!
+pub struct StaticLeak<T: 'static> {
+    inner: &'static T,
+    _ptr: Box<T>,
+}
+
+#[allow(dead_code)]
+impl<T> StaticLeak<T> {
+    /// Leaks a value T while keeping the ownership intact to make sure all memory gets freed as
+    /// soon as this gets dropped.
+    ///
+    /// With the help of unsafe code this circumvents borrow-checking as we "own an unowned value".
+    #[allow(unsafe_code)]
+    pub fn new(value: T) -> Self {
+        // Leak the inner value, from now on the value is not owned anymore. If we loose the
+        // reference, the value will still exist in memory, this could potentially cause a memory
+        // leak!
+        let inner = Box::leak(Box::new(value));
+
+        // Without the knowledge of the borrow-checker we "reclaim" the ownership and keep it
+        // inside the struct. Whenever the struct gets dropped, we can safely drop the Box and also
+        // free the used memory.
+        let _ptr = unsafe { Box::from_raw(inner) };
+
+        Self { inner, _ptr }
+    }
+
+    /// Returns a reference to the value T with 'static lifetime.
+    ///
+    /// Warning: The borrow-checker "assumes" that this value will exist until the end of the
+    /// program, but that might not be the case (see examples above)! Do only use it if you either:
+    ///
+    /// 1. Make sure this value will actually be used until the end of the program (it is truly
+    ///    'static).
+    ///
+    /// 2. Make sure you do not refer to it anymore _after_ you've dropped the `StaticLeak`
+    ///    instance.
+    pub fn unsafe_value(&self) -> &'static T {
+        self.inner
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{thread, time};
+
+    use rstest::rstest;
+
+    use p2panda_rs::schema::{FieldType, Schema, SchemaId};
+    use p2panda_rs::test_utils::fixtures::schema;
+
+    use super::{load_static_schemas, save_static_schemas, StaticLeak};
+
+    #[rstest]
+    fn save_and_load_schemas(#[from(schema)] schema_id: SchemaId) {
+        // Create and store schemas in global storage
+        let schema =
+            Schema::new(&schema_id, "Test schema", vec![("name", FieldType::String)]).unwrap();
+        let schemas = vec![schema];
+        save_static_schemas(&schemas);
+
+        // Load from global storage
+        let from_store = load_static_schemas();
+        assert_eq!(&schemas, from_store);
+    }
+
+    // #[test]
```

Comment:
I'm happy to remove this for now or add a `@TODO`, up to you :+1:

Reply:
We probably need more time than we have to figure this out and we don't even use it yet

Reply:
Interesting: I recently updated Rust and now this seems to pass. I want to try whether the error comes back when I go back to an old Rust version.

Reply:
Ok, I remove it.